### DISCUSSION TITLE
Frsky sbus failsafe/inav/Matek problems

### DISCUSSION THREAD

**Discussion by m-fuka** [iNavFlight/inav#7963] [Q&A]
I have 5 sets of aircraft in the configuration of Matek f405wing and f405wse with R9 receivers. 

After updating all Frsky R9 receivers, (custom) failsafe stopped working via sbus. (custom failsafe rx pwm output works fine)

  Even if I have set my own failsafe, fc matek evaluates failsafe HOLD !!!

When the transmitter is switched off, the fc Matek will be switched to rth due to the loss of the connection, but the own channels that I have preset will not be preset.

Matek channel 5 = modes
Transmitter off / failsafe on channel 5 = 100%
Matek mode nav rth on = 100%

I solved it with the representation of Frsky, allegedly it is not a mistake of Frsky but a mistake of misinterpretation of the sbus inav or matek.

The problem is mainly with through channels where I need to have my own values in case of connection failure.

I only have this problem with Frsky/Inav/Matek , for example with frsky/qgroundcontrol/pixhawk4 it works properly.

Has anyone solved the same problem?
Thanks.

**Comment by MrD-RC** [comment]
You should not be using the custom failsafe method. You should set failsafe on the receiver to **No Pulses** or **Hold**. 

SBUS has a failsafe bit that is interpreted by the flight controller. Custom channels should not be used. It is an out-dated method for protocols that did not have failsafe bits.

**Comment by m-fuka** [comment]
Ok, I understand that it is an outdated method, but for many years of use it did not disappoint me and especially the setup was very simple

The problem is:

  If I set the sbus HOLD - the pass-through channels remain in the last recorded position

If I set frsk sbus NO PULSE - the servo of the through channels will remain in the last recorded position again.

This is not permissible for through channels such as flaperons or brakes.
If they were activated before the outage (Multiplex Lentus 3000), the aircraft will fly on them during RTH failsafe.

It was working before the rx update
 
I'm sorry, let's not deal with the transition to failsafe, but the setting of pass-through channels so that after a signal failure they go to a position defined by me.

Now I have brake and flaperon servos connected to the pwm outputs of the Matek board, it will be easier for me to connect them directly to the pwm outputs of the receiver where pwm failsafe works.

It will probably be set in inav but I haven't been able to do that yet

**Comment by MrD-RC** [comment]
Actually, you raise a fair point. There should be setting in iNav for the failsafe positions of non-stabilised RC mixer lines. This should be set in iNav, not on the transmitter, as iNav is what does the mixing. The the failsafe bit is also still all that is needed.

**Comment by m-fuka** [comment]
The impossibility of setting positions also affects the flaperons mix and the flaperons / elevator mix set in inav

If you have flaperons activated when the connection fails, the flaperons and elevator areas will not return to the zero position.
The flaperons are active during the failsafe (red parachute).

For aircraft with complex mixing, the old proven method of self-failsafe transition was much simpler than "no pulses" or "hold". everything worked excellently. Now I have grounded 5 aircraft with Frsky / Matek / inav configuration due to this problem with sbus "custom" failsafe

I absolutely don't know why "custom" sbus doesn't work, where's the mistake? Frsky, Inav or Matek?

So far I have no choice but to get rid of complex mixes in inav, leave only AETR, failsafe to set NO Pulse for AETR and connect flaps directly to the pwm output of the receiver with "custom" failsafe setting. Or turn a 3 meter glider into a four channel.
